sdk/resourcehealth/azure-resourcemanager-resourcehealth/src/test/java/com/azure/resourcemanager/resourcehealth/ResourceHealthTests.java

```diff
@@ -0,0 +1,120 @@
+import com.azure.resourcemanager.compute.models.VirtualMachineSizeTypes;
+import com.azure.resourcemanager.resourcehealth.models.AvailabilityStateValues;
+import com.azure.resourcemanager.resourcehealth.models.AvailabilityStatus;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Random;
+
+public class ResourceHealthTests extends TestBase {
+    private static final Random RANDOM = new Random();
+
+    private static final Region REGION = Region.US_WEST3;
+    private static final String VM_NAME = "vm" + randomPadding();
+
+    private String resourceGroup = "rg" + randomPadding();
+
+
+    private static String randomPadding() {
+        return String.format("%05d", Math.abs(RANDOM.nextInt() % 100000));
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void resourceHealthTest() {
+        ComputeManager computeManager = ComputeManager
+            .configure().withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(new DefaultAzureCredentialBuilder().build(), new AzureProfile(AzureEnvironment.AZURE));
+
+        ResourceHealthManager resourceHealthManager = ResourceHealthManager
+            .configure().withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+            .authenticate(new DefaultAzureCredentialBuilder().build(), new AzureProfile(AzureEnvironment.AZURE));
+
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        boolean testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroup = testResourceGroup;
+        } else {
+            computeManager.resourceManager().resourceGroups().define(resourceGroup)
+                .withRegion(REGION)
+                .create();
+        }
+
+        try {
+            // create vm
+            VirtualMachine virtualMachine =
+                computeManager
+                    .virtualMachines()
+                    .define(VM_NAME)
+                    .withRegion(REGION)
+                    .withExistingResourceGroup(resourceGroup)
+                    .withNewPrimaryNetwork("10.0.0.0/28")
+                    .withPrimaryPrivateIPAddressDynamic()
+                    .withoutPrimaryPublicIPAddress()
+                    .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_18_04_LTS)
+                    .withRootUsername("azuser")
+                    .withRootPassword("Pa5$123456")
+                    .withSize(VirtualMachineSizeTypes.STANDARD_B1S)
+                    .create();
+
+            // get current availability status
+            AvailabilityStatus vmAvailabilityStatus = resourceHealthManager.availabilityStatuses().getByResource(virtualMachine.id());
+            while (!AvailabilityStateValues.AVAILABLE.equals(vmAvailabilityStatus.properties().availabilityState())) {
```

Comment:
Any idea why it is not AVAILABLE after VM create?

(or should we try `beginCreate` on above, as demonstrate this poll?)

Reply:
LGTM